lib/libmbedtls/core/sub.mk

```diff
@@ -1,3 +1,13 @@
 cflags-lib-$(CFG_CRYPTO_SIZE_OPTIMIZATION) += -Os
 
 srcs-y += tee_lmd_provider.c
+
+SRCS :=
+
+# Hashes
+SRCS += md5.c
+SRCS += sha1.c
+SRCS += sha256.c
+SRCS += sha512.c
+
+srcs-y += $(addprefix ../mbedtls/library/, $(SRCS))
```

Comment:
I'd prefer if we did this in `lib/libmbedtls/sub.mk`.
`lib/libmbedtls/sub.mk` will need to be restructured a bit. We could do it a bit like:
```
SRCS_CRYPTO := 0
SRCS_CRYPTO += aes.c
...
SRCS_X509 := 0
SRCS_X509 +=  certs.c
...
SRCS_TLS := 0
SRCS_TLS += debug.c
...
srcs-y += $(addprefix mbedtls/library/, $(SRCS_CRYPTO))
srcs-$(sm-$(ta-target)) += $(addprefix mbedtls/library/, $(SRCS_X509))
srcs-$(sm-$(ta-target)) += $(addprefix mbedtls/library/, $(SRCS_TLS))
```
It doesn't matter if a few too many files are included as it's the `lib/libmbedtls/include/mbedtls_config_kernel.h` and `lib/libmbedtls/include/mbedtls_config_uta.h` files that controls what's really compiled.

Reply:
> It doesn't matter if a few too many files are included as it's the `lib/libmbedtls/include/mbedtls_config_kernel.h` and `lib/libmbedtls/include/mbedtls_config_uta.h` files that controls what's really compiled.

By that logic, why not add all the files as it's done currently?

Reply:
> By that logic, why not add all the files as it's done currently

Fine with me, if we think it adds too much time to compiling we can address that then.